src/python/qeqhipster/simulator.py

```diff
@@ -1,115 +1,156 @@
-        )
-        os.putenv("TBBROOT", "/opt/intel/psxe_runtime_2019.3.199/linux/tbb")
-        os.putenv(
-            "PKG_CONFIG_PATH",
-            "/opt/intel/psxe_runtime_2019.3.199/linux/mkl/bin/pkgconfig",
-        )
-        os.putenv("I_MPI_ROOT", "/opt/intel/psxe_runtime_2019.3.199/linux/mpi")
-
+    @compatible_with_old_type(
+        old_type=OldCircuit, translate_old_to_wip=new_circuit_from_old_circuit
+    )
     def run_circuit_and_measure(self, circuit, n_samples=None, **kwargs):
         if n_samples is None:
             n_samples = self.n_samples
         wavefunction = self.get_wavefunction(circuit)
         return Measurements(sample_from_wavefunction(wavefunction, n_samples))
 
+    @compatible_with_old_type(
+        old_type=OldCircuit, translate_old_to_wip=new_circuit_from_old_circuit
+    )
     def get_exact_expectation_values(self, circuit, qubit_operator, **kwargs):
         self.number_of_circuits_run += 1
         self.number_of_jobs_run += 1
         circuit = make_circuit_qhipster_compatible(circuit)
-        save_circuit(circuit, "./temp_qhipster_circuit.json")
-        if isinstance(qubit_operator, SymbolicOperator):
-            save_symbolic_operator(qubit_operator, "./temp_qhipster_operator.json")
-        else:
-            raise Exception(
-                "Unsupported type: "
-                + type(qubit_operator)
-                + "QHipster works only with openfermion.SymbolicOperator"
+
+        with tempfile.TemporaryDirectory() as dir_path:
+            operator_json_path = os.path.join(dir_path, "temp_qhipster_operator.json")
+            operator_txt_path = os.path.join(dir_path, "temp_qhipster_operator.txt")
+            circuit_txt_path = os.path.join(dir_path, "temp_qhipster_circuit.txt")
+            expectation_values_json_path = os.path.join(
+                dir_path, "expectation_values.json"
+            )
+
+            if not isinstance(qubit_operator, SymbolicOperator):
+                raise TypeError(
+                    f"Unsupported type: {type(qubit_operator)} QHipster works only with "
+                    "openfermion.SymbolicOperator"
+                )
+
+            save_symbolic_operator(qubit_operator, operator_json_path)
+
+            with open(circuit_txt_path, "w") as qasm_file:
+                qasm_file.write(convert_to_simplified_qasm(circuit))
+
+            subprocess.call(
+                [
+                    "/app/json_parser/qubitop_to_paulistrings.o",
+                    operator_json_path,
+                ]
+            )
+            # Run simulation
+            subprocess.call(
+                [
+                    "/app/zapata/zapata_interpreter_no_mpi_get_exp_vals.out",
+                    circuit_txt_path,
+                    str(self.nthreads),
+                    operator_txt_path,
+                    expectation_values_json_path,
+                ],
+                env=PSXE_ENVS,
```

Comment:
I'm glad y'all figured this out. I remember playing around with it, but couldn't figure out how to make it work. Looks much cleaner now.